projects/schematics/src/modernize-app-migrated-from-68-to-221119/csr-and-ssr/update-angular-json-for-application-builder.ts

```diff
@@ -0,0 +1,67 @@
+import { Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { getWorkspace } from '../../shared/utils/workspace-utils';
+
+/**
+ * Updates the Angular configuration file to new Angular v17 standards.
+ *
+ * Configures the new Angular application builder:
+ * '@angular-devkit/build-angular:application',
+ * updates output paths, and removes obsolete build options.
+ */
+export function updateAngularJsonForApplicationBuilder(): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    context.logger.info('⏳ Updating angular.json for application builder...');
+
+    const { workspace, path } = getWorkspace(tree);
+    const project = workspace.projects[Object.keys(workspace.projects)[0]];
+
+    if (!project) {
+      throw new Error('No project found in workspace');
+    }
+
+    const buildTarget = project.architect?.build as any;
+    if (!buildTarget) {
+      throw new Error('No build target found in project configuration');
+    }
+
+    // Update builder
+    buildTarget.builder = '@angular-devkit/build-angular:application';
+
+    // Update options
+    const options = buildTarget.options as any;
+    if (options) {
+      // Update outputPath
+      if (
+        typeof options.outputPath === 'string' &&
+        options.outputPath.endsWith('/browser')
+      ) {
+        options.outputPath = options.outputPath.replace(/\/browser$/, '');
+      } else {
+        throw new Error('Could not update "outputPath" in angular.json');
+      }
```

Comment:
`major:` This logic will throw an error for OOTB CSR application resulting with no changes in `angular.json`. Let's add to the `if` statement logic to recognize if app uses SSR.

Reply:
We can also consider moving this step to `update-angular-json-for-ssr.ts` file as it's not common for both types of apps.

Reply:
CSR QA outcome:
![image](https://github.com/user-attachments/assets/82c41ff4-fb0b-4316-8f96-245a723fa346)

Reply:
Thanks. good catch. 
Then I'll fix also the Markdown guide in `develop` branch - because it has the same mistake (it assumed this manual step for both SSR **and CSR** apps).

Reply:
Created a separate PR to fix the migration docs in `develop`: https://github.com/SAP/spartacus/pull/19960

Reply:
Also now pushed a fix in this PR:
- [updated schematics logic](https://github.com/SAP/spartacus/pull/19957/commits/84dc472824250fa5263c6613a8a08b9b931c80ba)
- [updated the copied mardown](https://github.com/SAP/spartacus/pull/19957/commits/17ec14a76b0a66b58e82cf973f84db8a1540e0e3) similar to changes in the PR https://github.com/SAP/spartacus/pull/19960